packages/odie-client/src/components/send-message-input/index.tsx

```diff
@@ -1,26 +1,115 @@
 import { AttachmentButton } from './attachment-button';
 import { ResizableTextarea } from './resizable-textarea';
 
 import './style.scss';
 
+const getFileType = ( file: File ) => {
+	if ( file.type.startsWith( 'image/' ) ) {
+		return 'image-placeholder';
+	}
+
+	return 'text';
+};
+
+const getPlaceholderAttachmentMessage = ( file: File ) => {
+	return zendeskMessageConverter( {
+		role: 'user',
+		type: getFileType( file ),
+		displayName: '',
+		text: '',
+		id: String( new Date().getTime() ),
+		received: new Date().getTime(),
+		source: { type: 'web', id: '', integrationId: '' },
+		mediaUrl: URL.createObjectURL( file ),
+	} );
+};
+
 export const OdieSendMessageButton = () => {
 	const divContainerRef = useRef< HTMLDivElement >( null );
 	const inputRef = useRef< HTMLTextAreaElement >( null );
 	const attachmentButtonRef = useRef< HTMLElement >( null );
-	const { trackEvent, chat } = useOdieAssistantContext();
+	const { trackEvent, chat, addMessage, isUserEligibleForPaidSupport } = useOdieAssistantContext();
 	const sendMessage = useSendChatMessage();
 	const isChatBusy = chat.status === 'loading' || chat.status === 'sending';
 	const [ isMessageSizeValid, setIsMessageSizeValid ] = useState( true );
 	const [ submitDisabled, setSubmitDisabled ] = useState( true );
 
+	const { data: authData } = useAuthenticateZendeskMessaging(
+		isUserEligibleForPaidSupport,
+		'messenger'
+	);
+	const { zendeskClientId } = useSelect( ( select ) => {
+		const helpCenterSelect: HelpCenterSelect = select( HELP_CENTER_STORE );
+		return {
+			zendeskClientId: helpCenterSelect.getZendeskClientId(),
+		};
+	}, [] );
+	const inferredClientId = chat.clientId ? chat.clientId : zendeskClientId;
+
+	const { isPending: isAttachingFile, mutateAsync: attachFileToConversation } =
+		useAttachFileToConversation();
+
+	const handleFileUpload = useCallback(
+		async ( file: File ) => {
+			if ( file.type.startsWith( 'image/' ) ) {
+				if ( authData && chat.conversationId && inferredClientId && file ) {
+					attachFileToConversation( {
+						authData,
+						file,
+						conversationId: chat.conversationId,
+						clientId: inferredClientId,
+					} ).then( () => {
+						addMessage( getPlaceholderAttachmentMessage( file ) );
+						trackEvent( 'send_message_attachment', { type: file.type } );
+					} );
+				}
+			} else {
+				addMessage( ODIE_WRONG_FILE_TYPE_MESSAGE );
+			}
+		},
+		[
+			authData,
+			chat.conversationId,
+			inferredClientId,
+			attachFileToConversation,
+			addMessage,
+			trackEvent,
+		]
+	);
+
+	const onFilesDrop = ( files: File[] ) => {
+		const file = files?.[ 0 ];
+		if ( file && file.type.startsWith( 'image/' ) ) {
+			handleFileUpload( file );
+		}
+	};
+
+	const onPaste = ( event: React.ClipboardEvent ) => {
+		const items = event.clipboardData.items;
+		const file = items?.[ 0 ]?.getAsFile();
+		if ( file && file.type.startsWith( 'image/' ) ) {
+			event.preventDefault();
+			handleFileUpload( file );
+		}
```

Comment:
I noticed that pasting a file of the wrong format works but pastes the file name, which can easily be confused for the file itself. 

We should pass the file to `handleFileUpload` without checking its type here.